src/MarketPlace.sol

```diff
@@ -207,50 +177,22 @@ contract MarketPlace {
     /// @param u address of an underlying asset
     /// @param m maturity (timestamp) of the market
     /// @param a address of the new principal token
-    /// @param h a supplementary address (apwine needs a router, element needs a vault, sense needs interest bearing asset)
-    /// @param sensePeriphery address of the Sense periphery contract that must be approved by the lender
+    /// @param adapter address of the protocol's adapter contract
     /// @return bool true if the principal set, false otherwise
     function setPrincipal(
```

Comment:
Same thing regarding approvals here

Reply:
We will still need the approvals for the specific routers / protocol specific adapters. However, because of the nature of modularization, we won't be able to hard-code the exact `p` value to the protocol at the time of `createMarket`. Instead, we'll have to either include an `approve` method inside the adapter for the contract to hook into, or use the `admin` approval method to make this happen. I'll make a separate ticket for us to handle this.